app/client/views/settings/data/data.js

```diff
@@ -1,70 +1,138 @@
-import FileSaver from 'file-saver';
+import FileSaver from "file-saver";
 
-Template.dataSettings.onCreated(function () {
+const showAlert = () => {
+  const templateInstance = Template.instance();
+  templateInstance.shouldShowAlert.set(true);
+
+  setTimeout(() => {
+    templateInstance.shouldShowAlert.set(false);
+  }, 3000);
+};
+
+const _clear = () => {
+  const templateInstance = Template.instance();
+
+  templateInstance.file.set(null);
+  templateInstance.isFileInvalid.set(false);
+
+  document.getElementById("inputFile").value = "";
+};
+
+Template.dataSettings.onCreated(function() {
   // Get reference to template instance
   const templateInstance = this;
 
   // 'fetching data' variable for UI state
   templateInstance.fetchingData = new ReactiveVar(false);
 
   templateInstance.isFileInvalid = new ReactiveVar(false);
+  templateInstance.shouldShowAlert = new ReactiveVar(false);
+  templateInstance.file = new ReactiveVar(null);
+  templateInstance.isRespError = new ReactiveVar(false);
+  templateInstance.respMessage = new ReactiveVar(null);
 });
 
 Template.dataSettings.events({
-  'click #export-data' (event, templateInstance) {
+  "click #export-data"(event, templateInstance) {
     // Set 'fetching data' variable to true
     templateInstance.fetchingData.set(true);
 
     // Export all data
-    Meteor.call('exportAllData', function (error, exportData) {
+    Meteor.call("exportAllData", function(error, exportData) {
       if (!error) {
         // Convert export data to string
-        const dataString = JSON.stringify(exportData)
+        const dataString = JSON.stringify(exportData);
 
         // Create binary object from data string
-        const exportBlob = new Blob([dataString], {type: 'application/json;charset=utf-8'});
+        const exportBlob = new Blob([dataString], {
+          type: "application/json;charset=utf-8"
+        });
 
         // Trigger save as prompt
-        FileSaver.saveAs(exportBlob, 'GeriLife-export.json');
+        FileSaver.saveAs(exportBlob, "GeriLife-export.json");
 
         // hide spinner
         templateInstance.fetchingData.set(false);
       }
     });
   },
 
-  "change .file-upload-input": function(event, templateInstance){
-    
+  "change .file-upload-input": function(event, templateInstance) {
     var file = event.currentTarget.files[0];
-    if(file.type!=="application/json"){
+    templateInstance.file.set(file);
+    if (file.type !== "application/json") {
+      showAlert();
       templateInstance.isFileInvalid.set(true);
-      return
-    } 
+
+      return;
+    }
+    templateInstance.isFileInvalid.set(false);
+  },
+
+  "click #importData": function(event, templateInstance) {
     var reader = new FileReader();
-    
+
     reader.onload = function(fileLoadEvent) {
-      console.log(reader.result)
-       Meteor.call('file-upload', reader.result);
+      Meteor.call("JSONFileImport", reader.result, function(data) {
+        Session.set("JSONFileImportResult", data);
```

Comment:
Lets use a template instance reactive variable here, rather than a session variable. That way, the reactive variable is cleaned up when the template is destroyed.

Basically, use session variables sparingly, only for data that needs to exist across routes/templates or page reloads.